### PR TITLE
Add new user domain permission check functions

### PR DESCRIPTION
**Pull request by Amtron**

### CODE REVIEW

File: typescript/src/constants/roles.ts [hunk 1]
```diff
@@ -6,14 +6,22 @@ export enum Role {
   SUPPLIER = 4,
   CLIENT = 5,
   MANAGER = 6,
-  ACCOUNTANT = 7
+  ACCOUNTANT = 7,
```
Comment (DanielCollins) [inline typescript/src/constants/roles.ts]: doesn't seem related

File: typescript/src/constants/roles.ts [hunk 2]
```diff
@@ -6,14 +6,22 @@ export enum Role {
...
+];
+
+export const MANAGEMENT_TEAM = [
+  ...DOMAIN_MANAGERS,
   Role.SALES,
   Role.DESIGNER,
-  Role.SUPPLIER,
-  Role.MANAGER,
-  Role.ACCOUNTANT
+  Role.ACCOUNTANT,
```
Comment (DanielCollins) [inline typescript/src/constants/roles.ts]: doesn't seem related

File: typescript/src/constants/roles.ts [hunk 3]
```diff
@@ -6,14 +6,22 @@ export enum Role {
...
+];
+
+export const BUSINESS_ACCOUNTS = [
+    ...MANAGEMENT_TEAM,
+    Role.SUPPLIER
```
Comment (DanielCollins) [inline typescript/src/constants/roles.ts]: you've added trailing `,` to every exiting array, but not to the new one you created?

obviously not a major issue but i will prefer if every pr is smaller and more consistent.

File: typescript/src/entities/user.ts [hunk 4]
```diff
@@ -246,7 +250,15 @@ export class User extends Entity {
...
+  public isDomainManager(domain: Domain) {
+    return DOMAIN_MANAGERS.includes(this.roleInDomain(domain));
+  }
+
   public isManagementTeam(domain: Domain) {
     return MANAGEMENT_TEAM.includes(this.roleInDomain(domain));
   }
+
+  public isNotClient(domain: Domain) {
+    return BUSINESS_ACCOUNTS.includes(this.roleInDomain(domain));
```
Comment (DanielCollins) [inline typescript/src/entities/user.ts]: i have to note that merchicrm cannot import this as of `develop` -- webpack or whatever is not setup to understand `includes`

File: typescript/src/constants/roles.ts (hunk 3 above)
Comment (Amtron) [reply typescript/src/constants/roles.ts]: added the final comma.

File: typescript/src/entities/user.ts (hunk 4 above)
Comment (Amtron) [reply typescript/src/entities/user.ts]: have to talk about @s3341458 about this.
